File: Java/409. Longest Palindrome.java
class Solution {
    public int longestPalindrome(String s) {
        int[] map = new int[128];
        for(char c: s.toCharArray()){
            map[c]++;
        }
        int r = 0, t = 0;
        for(int v: map){
            if(v % 2 == 1){t = 1;}
            r += v / 2 * 2;
        }
        return r + t;
    }
}

File: Python/409. Longest Palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        r, t = 0, 0
        for i in Counter(s).values():
            if i % 2 == 1: t = 1
            r += (i // 2) * 2
        return r + t
